app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -3758,6 +3855,9 @@ private void notifyProgressUpdateToListeners(final int currentProgress,
         if (activityListener != null) {
             activityListener.onProgressUpdate(currentProgress, duration, bufferPercent);
         }
+        if (areSegmentsVisible) {
+            segmentAdapter.selectSegmentAt(getNearestStreamSegmentPosition(currentProgress));
+        }
```

Comment:
This is not a listener, so I don't think this piece of code should be here, but maybe I'm wrong

Reply:
In fact, you're right, I'll move it to the "parent" method (`onUpdateProgress`), should be better.